subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/Checkstyle.java

```diff
@@ -159,6 +160,7 @@ public void run() {
      * executed.</p>
      */
     @Override
+    @IgnoreEmptyDirectories
```

Comment:
`Checkstyle` is a source task, so it already has `@IgnoreEmptyDirectories`.

Reply:
In that case is `@PathSensitive(PathSensitivity.RELATIVE)` also redundant? Should that be removed?

Reply:
`SourceTask` has absolute path sensitivity for historical reasons. So it can't be removed.